### DISCUSSION TITLE
Issue with Helm hooks in base chart

### DISCUSSION THREAD

**Discussion by m8-t** [argoproj/argo-cd#16263] [Q&A]
Hey,

right now I am a little bit struggling with a Helm Chart provided by a 3rd party. I created an umbrella chart that has the 3rd party helm chart as dependency. I added some additional stuff for automation.

My first install runs just fine as expected. I wanted to test what happens if I do a full sync without any changes in the code base. I expected nothing will happen but I noticed that Argo CD deletes some hook annotated resources. Hooks set are these (out of the base chart):

```
    "helm.sh/hook": "pre-install,pre-upgrade"
    "helm.sh/resource-policy": keep
```
Afaik Argo CD does respect Helm hooks and I dont need to rewrite them to Argo Hooks as long I am not mixing Argo and Helm Hooks. My research ended up in "Argo CD uses Helm just as a templating engine" - this is fine in most of the cases I guess :). But as far as I can see Argo CD executes the Hooks that are hard coded inside the base chart on every sync operation. I do not want to rewrite the base chart, so I am looking for a solution to not touch the base chart at all and maybe overwrite it somehow with my umbrella chart. Is there a neat workaround for this? Any hint?

Thanks

**Comment by m8-t** [comment]
As of Argo CD 2.9 it supports Kustomize Patch fields in the Application CRD - I will try this out to overwrite these hooks.

**Reply by marinkovicpetar** [reply]
@dan-m8t did you maybe find some solution to this problem, as I also have the same issue

**Comment by raweber42** [comment]
Any updates here?
